File: Payload/Ruby/CGI/OneShell/db_init.rb
def main
  checks = {
    'MySQLi' => false,
    'MySQL' => false,
    'PDO' => false,
    'PDO MySQL' => false,
    'PDO PostgreSQL' => false,
    'PDO SQLite' => false,
    'PostgreSQL' => false,
    'SQLite3' => true,
    'Redis' => false,
    'MongoDB' => false,
    'Oracle (OCI8)' => false,
    'Microsoft SQL Server' => true,
    'ODBC' => true
  }

  checks.each do |db, available|
    status = available ? 1 : 0
    print "#{db}:#{status},"
  end

  return
end

main()

File: Payload/Perl/CGI/OneShell/db_init.pl
use strict;
use warnings;

my %checks = (
    'MySQLi (DBI::mysql)' => 'DBD::mysql',
    'PDO MySQL (DBI::mysql)' => 'DBD::mysql',
    'PDO PostgreSQL (DBI::Pg)' => 'DBD::Pg',
    'PDO SQLite (DBI::SQLite)' => 'DBD::SQLite',
    'PostgreSQL' => 'DBD::Pg',
    'SQLite3' => 'DBD::SQLite',
    'Redis' => 'Redis',
    'MongoDB' => 'MongoDB',
    'Oracle (OCI8)' => 'DBD::Oracle',
    'Microsoft SQL Server' => 'DBD::ODBC',
    'ODBC' => 'DBD::ODBC',
);

for my $db (keys %checks) {
    my $module = $checks{$db};

    my $available = eval "require $module; 1" ? 1 : 0;

    print "$db:$available,";
}